### PR TITLE
fix(docs): Fix some inaccuracies in Docker docs

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

I noticed we have some inaccuracies in the docs for Docker.

Depends-On: #6729
(Required fix for CI to pass.)

## Solution

- Fix dead links.
- Fix the `docker build` command.
- Remove a redundant subsection as @gustavovalverde suggested.

## Review

@gustavovalverde 

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: book/src/user/docker.md [hunk 1]
```diff
@@ -18,27 +18,12 @@ docker run --detach zfnd/zebra:1.0.0-rc.8
...
 ```shell
 git clone --depth 1 --branch v1.0.0-rc.8 https://github.com/ZcashFoundation/zebra.git
-docker build --file docker/Dockerfile --target runtime --tag zebra:local
+docker build --file docker/Dockerfile --target runtime --tag zebra:local .
 docker run --detach zebra:local
 ```
 
-### Alternatives
-
-See the Zebra [build instructions](https://github.com/ZcashFoundation/zebra#build-instructions).
```
Comment (gustavovalverde) [inline book/src/user/docker.md:27]: I would keep this section, but referencing the actual link in the README https://github.com/ZcashFoundation/zebra#building-zebra

**Review by gustavovalverde — CHANGES_REQUESTED** [review]

File: book/src/user/docker.md (hunk 1 above)
Comment (upbqdn) [reply book/src/user/docker.md:27]: Done in [c8fe01f](https://github.com/ZcashFoundation/zebra/pull/6723/commits/c8fe01fd467c376f988121dade294639aff6257e).

**Review by gustavovalverde — APPROVED** [review]

**Comment by teor2345** [comment]
@mergifyio update

**Review by teor2345 — APPROVED** [review]
Needed the CI fix in PR #6729 to merge

**Pull request merged**